fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateSort.java

```diff
@@ -57,21 +59,24 @@ public Plan visit(Plan plan, Boolean pruneSort) {
     @Override
     public Plan visitLogicalSort(LogicalSort<? extends Plan> sort, Boolean eliminateSort) {
         if (eliminateSort) {
-            return visit(sort.child(), true);
+            return visit(sort.child(), eliminateSort);
         }
-        return visit(sort, true);
+        return visit(sort, eliminateSort);
     }
```

Comment:
why pass `eliminateSort` rather than true

Reply:
have fixed it